autoload/lsp/diag.vim

```diff
@@ -41,6 +46,15 @@ def DiagsRefreshSigns(lspserver: dict<any>, bnr: number)
     signs->add({id: 0, buffer: bnr, group: 'LSPDiag',
 				lnum: lnum,
 				name: DiagSevToSignName(diag.severity)})
+
+    if has('patch-9.0.1157') && opt.lspOptions.showDiagWithVirtualText
+        prop_add( lnum, 0, {
+            'bufnr': bnr,
+            'type': 'LspDiagVirtualText',
+            'text': '┌─ ' .. diag.message,
+            'text_align': 'above',
+            'text_padding_left': diag.range.start.character})
+    endif
   endfor
```

Comment:
Can you modify the DiagsHighlightDisable() function to remove the virtual text?

Reply:
in https://github.com/yegappan/lsp/pull/187/commits/fdcb549003e49a48f5846911f25afaa4b2bef42d connect virtual text display to the autoHighlightDiags option as result we have on / off all Diags highlight and virtual text with :LspDiagHighlightEnable() :LspDiagHighlightDisable()